Marlin/ultralcd.cpp

```diff
@@ -830,14 +848,13 @@ static void _lcd_move(const char* name, AxisEnum axis, int min, int max) {
     if (min_software_endstops) NOLESS(current_position[axis], min);
     if (max_software_endstops) NOMORE(current_position[axis], max);
     encoderPosition = 0;
-    if (movesplanned() <= 3)
-      line_to_current(axis);
+    manual_move_to_current(axis);
     lcdDrawUpdate = 1;
   }
   if (lcdDrawUpdate) lcd_implementation_drawedit(name, ftostr31(current_position[axis]));
   if (LCD_CLICKED) {
-      line_to_current(axis);
-      lcd_goto_menu(lcd_move_menu_axis);
+    manual_move_to_current(axis);
+    lcd_goto_menu(lcd_move_menu_axis);
   }
```

Comment:
The trick was to enqueue the last move unconditionally, when we leave the menu. Because we used (movesplanned() <= 3) before, there is always enough headroom to push one move.
So in your version you can safely drop `manual_move_to_current(axis);` here.